runtime/shared_common/OSCache.cpp

```diff
@@ -1313,8 +1314,19 @@ SH_OSCache::getCacheNameAndLayerFromUnqiueID(J9JavaVM* vm, const char* cacheDirN
 	char nameWithVGenCopy[J9SH_MAXPATH];
 	memset(nameWithVGenCopy, 0, J9SH_MAXPATH);
 	strncpy(nameWithVGenCopy, cacheNameWithVGenStart, cacheNameWithVGenEnd - cacheNameWithVGenStart);
+	
+	J9PortShcVersion versionData;
+	getValuesFromShcFilePrefix(PORTLIB, nameWithVGenCopy, &versionData);
+	UDATA prefixLen = J9SH_VERSION_STRING_LEN + 1;
+	if (J9PORT_SHR_CACHE_TYPE_NONPERSISTENT == versionData.cacheType) {
+#if defined(WIN32)
```

Comment:
It would be useful to have comments that explain the differences in the multiple cases.

Reply:
Comments added